### PR TITLE
StorageEngine - add ability to read commands

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

**Comment by davidegrohmann** [comment]
I don't really like making all the command readers in a single hierarchy, i.e., 2.2.4->2.2->2.1->2.0->1.9. Maintaining that hierarchy in the correct state is gonna be complicated and error prone. Let's think a bit more about that, moreover it can be solved later and not necessary in this PR.

**Comment by tinwelint** [comment]
You're right about the hierarchy, I gave that a thought as well. What's clear about it though is that you can see exactly what changed between versions... but that could probably be achieved in another design as well.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+        DynamicRecord record = new DynamicRecord( id );
+        record.setInUse( inUse, type );
+        if ( inUse )
+        {
+            record.setStartRecord( (inUseFlag & Record.FIRST_IN_CHAIN.byteValue()) != 0 );
+            int nrOfBytes = channel.getInt();
+            assert nrOfBytes >= 0 && nrOfBytes < ((1 << 24) - 1) : nrOfBytes
+                                                                   +
+                                                                   " is not valid for a number of bytes field of " +
+                                                                   "a dynamic record";
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java]: this formatting looks weird...

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+            long nextBlock = channel.getLong();
+            assert (nextBlock >= 0 && nextBlock <= (1l << 36 - 1))
+                   || (nextBlock == Record.NO_NEXT_BLOCK.intValue()) : nextBlock
+                                                                       +
+                                                                       " is not valid for a next record field of " +
+                                                                       "a dynamic record";
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java]: this formatting looks weird...

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java]: Unchanged though, but could might as well reformat those files

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java (hunk 2 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogCommandReaderV2_2.java]: Unchanged though, but could might as well reformat those files

**Comment by davidegrohmann** [comment]
@tinwelint needs rebase

**Comment by tinwelint** [comment]
@davidegrohmann ready to merge now?

**Pull request merged**